File: js/src/member.js
class Member {
  constructor(user) {
    this.user = user;
  }

  // Extraction
  address() {
    const { address1, address2 } = this.user.data;
    return `${address1}\n${address2}`;
  }

  name() {
      return this.data.name;
  }
}

module.exports = Member;

File: rust/src/member.rs
use db::UserData;

pub struct Member {
    data: UserData,
}

impl Member {
    pub fn new(data: UserData) -> Member {
        Member { data }
    }

    pub fn name(&self) -> String {
        self.data.name.clone()
    }

    pub fn address(&self) -> String {
        let data = &self.data;
        format!("{}\n{}", &data.address1, &data.address2)
    }
}
